chainerx_cc/chainerx/native/native_device/linalg.cc

```diff
@@ -194,5 +284,131 @@ class NativeInverseKernel : public InverseKernel {
+                u = Empty(Shape{mn, m}, dtype, device);
+                vt = Empty(Shape{mn, n}, dtype, device);
+            }
+        } else {
+            u = Empty(Shape{0}, dtype, device);
+            vt = Empty(Shape{0}, dtype, device);
+        }
+
+        Array s = Empty(Shape{mn}, dtype, device);
+
+        auto svd_impl = [&](auto pt) -> std::tuple<Array, Array, Array> {
+            using T = typename decltype(pt)::type;
+
+            auto x_ptr = static_cast<T*>(internal::GetRawOffsetData(x));
+            auto s_ptr = static_cast<T*>(internal::GetRawOffsetData(s));
+            auto u_ptr = static_cast<T*>(internal::GetRawOffsetData(u));
+            auto vt_ptr = static_cast<T*>(internal::GetRawOffsetData(vt));
+
+            char job;
+            if (compute_uv) {
+                job = full_matrices ? 'A' : 'S';
+            } else {
+                job = 'N';
+            }
+
+            Array iwork = Empty(Shape{8 * mn}, Dtype::kInt64, device);
+            auto iwork_ptr = static_cast<int*>(internal::GetRawOffsetData(iwork));
+
+            int info;
+            int buffersize = -1;
+            T work_size;
+            Gesdd(job, m, n, x_ptr, m, s_ptr, u_ptr, m, vt_ptr, n, &work_size, buffersize, iwork_ptr, &info);
+            buffersize = static_cast<int>(work_size);
+
+            Array work = Empty(Shape{buffersize}, dtype, device);
+            auto work_ptr = static_cast<T*>(internal::GetRawOffsetData(work));
+
+            Gesdd(job, m, n, x_ptr, m, s_ptr, u_ptr, m, vt_ptr, n, work_ptr, buffersize, iwork_ptr, &info);
+
+            if (info != 0) {
+                throw ChainerxError{"Unsuccessful gesdd (SVD) execution. Info = ", info};
+            }
+
+            if (trans_flag) {
+                return std::make_tuple(std::move(u.Transpose()), std::move(s), std::move(vt.Transpose()));
+            } else {
+                return std::make_tuple(std::move(vt), std::move(s), std::move(u));
+            }
+        };
+
+        return VisitFloatingPointDtype(dtype, svd_impl);
+#else  // CHAINERX_LAPACK_AVAILABLE
+        (void)a;  // unused
+        (void)full_matrices;  // unused
+        (void)compute_uv;  // unused
+        throw ChainerxError{"LAPACK is not linked to ChainerX."};
+#endif  // CHAINERX_LAPACK_AVAILABLE
+    }
+};
+
+CHAINERX_NATIVE_REGISTER_KERNEL(SVDKernel, NativeSVDKernel);
+
+class NativePseudoInverseKernel : public PseudoInverseKernel {
+public:
+    void Call(const Array& a, const Array& out, float rcond = 1e-15) override {
+        Device& device = a.device();
+        device.CheckDevicesCompatible(a, out);
+
+        CHAINERX_ASSERT(a.ndim() == 2);
+
+        Array u{};
+        Array s{};
+        Array vt{};
+
+        std::tie(u, s, vt) = device.backend().CallKernel<SVDKernel>(a, false, true);
+
+        Array cutoff = rcond * s.Max();
+        Array cutoff_indices = s <= cutoff;
+
+        Array sinv = 1.0 / s;
+        sinv = Where(cutoff_indices, 0, sinv);
```

Comment:
Can you use `IfLessElse`? It seems like the argument types match if you can convert `cutoff_indices` to a `Scalar`.

Reply:
It is `sinv = s <= cutoff ? 0 : sinv`, technically it not the same as `IfLessElse`.
There is also `IfGreaterElse` for  `sinv = s > cutoff ? sinv : 0` but in this case argument types do not match.

Reply:
Okay, then let's leave it as is.